NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# Ah edit tag 16

# Description

As an admin I would like to be able to modify a Tag so that I can rephrase the name if I think of something more appropriate.
Given the user is viewing the Tag list
When they select the option to edit an Tag
Then the user should be directed to a form and given the ability to change the Tag's name
Given the user is finished updating the Tag information
When they click the Save button
Then the updated Tag should be saved to the database
And the user should be redirected to the new Tag List page
Given the user has decided not to edit the Tag
When they click the Cancel button
Then the user should be redirected back to the list page



**Created the following files and methods:**

- EditTag.js
- editTag()


**Related Ticket(s)**
https://trello.com/c/bfs0vsa2

#16 Edit a Tag

**Type of Change**


# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `AH-editTag-16` as one of the available branches
- `git checkout AH-editTag-16` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin AH-editTag-16`
- `Start TabloidMVC.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)


- You will be presented with a login screen. Enter the email foo@bar.com and click "Login" (no password required)

- You will be presented with the welcome screen that simply says "hello"

- Click the Tag Management link in the topside header

-You should see a table that contains all Tags from your database as well as their corresponding Id. This table should be in alphabetical order by Tag Name

-To the right of  each item on the list you should see an “Edit” button

-Click said button on the tag named ”Funny”

-You will then be redirected to a form “/tag/edit/3”

-Enter “Test” into the input field

-Click “Save” button

-You should then be redirected to the “/tag” list, which should include your newly edited “Test” tag.

-Run this SQL query:

```
SELECT * FROM Tag;
```
 to confirm that your new tag has been added to the database.


# Checklist:
